azkaban-common/src/test/java/azkaban/user/FileWatcherTest.java

```diff
@@ -73,14 +72,33 @@ public void registerAndTake() throws Exception {
 
     final WatchKey key = this.fileWatcher.take();
     final List<WatchEvent<?>> events = this.fileWatcher.pollEvents(key);
-    // depending on the OS & file system there may be at least 1 or 2 events even with 1 write()
```

Comment:
Did you somehow conclude that this isn't true actually?

Reply:
I would say the tolerance for more than 1 events should be kept.

> The implementation that observes events from the file system is intended to map directly on to the native file event notification facility where available, or to use a primitive mechanism, such as polling, when a native facility is not available. Consequently, many of the details on how events are detected, their timeliness, and whether their ordering is preserved are highly implementation specific. **For example, when a file in a watched directory is modified then it may result in a single ENTRY_MODIFY event in some implementations but several events in other implementations**.

This is from https://docs.oracle.com/javase/7/docs/api/java/nio/file/WatchService.html

Reply:
Yes -- it *is* kept. The check, as implemented, verifies that there will be *at least one* event. This is because `anyMatch()` will return `true` is there are more than 1 matching events. So the old additional check on the number of events becomes redundant. Or am I missing something? Thanks for keeping an eye on this, @juhoautio.

Reply:
Yes, sorry. anyMatch works the same as the previous code, and is cleaner. I actually wanted to say that the comment should be kept about why anyMatch is used here. Intuitively one might expect there to be just one event about file modification, and even more so if the developer’s own system only triggers one.

Reply:
OK, I see what you are saying. We may keep the note of the fact that even with one write to one file we may get multiple events. It will be even more relevant if I refactor this test to use a unique dir. Thanks for pointing that out.